common/lib/capa/capa/tests/test_capa_problem.py

```diff
@@ -0,0 +1,329 @@
+"""
+Test capa problem.
+"""
+import unittest
+
+from . import new_loncapa_problem
+
+
+class CAPAProblemTest(unittest.TestCase):
+    """ CAPA problem related tests"""
+
+    def test_label_and_description_inside_responsetype(self):
+        """
+        Verify that
+        * label is extracted
+        * description_ids are constructed
+        * <label> tag is removed to avoid duplication
+
+        This is the case when we have a problem with single question or
+        problem with multiple-questions separated as per the new format.
+        """
+        xml = """
+        <problem>
+            <choiceresponse>
+                <label>Select the correct synonym of paranoid?</label>
+                <description>Only the paranoid survive.</description>
+                <checkboxgroup>
+                    <choice correct="true">over-suspicious</choice>
+                    <choice correct="false">funny</choice>
+                </checkboxgroup>
+            </choiceresponse>
+        </problem>
+        """
+        problem = new_loncapa_problem(xml)
+        self.assertEqual(
+            problem.problem_data,
+            {
+                '1_2':
+                {
+                    'description_ids': 'description_1_2_1',
+                    'label': 'Select the correct synonym of paranoid?',
+                    'descriptions': {'description_1_2_1': 'Only the paranoid survive.'}
+                }
+            }
+        )
+        self.assertEqual(len(problem.tree.xpath('//label')), 0)
+
+    def test_label_attribute_only(self):
+        """
+        Verify that label is extracted and <p> tag with question
+        text is removed when label attribute is set on inputtype.
+        """
+        question = "Once we become predictable, we become ______?"
+        xml = """
+        <problem>
+            <p>Be sure to check your spelling.</p>
+            <p>{}</p>
+            <stringresponse answer="vulnerable" type="ci">
+                <textline label="{}" size="40"/>
```

Comment:
I see.. This label attribute does make sense, please ignore such existence from the list of files.